Distiller-server/src/main/java/com/mapr/distiller/server/processors/SystemCpuRecordProcessor.java

```diff
@@ -11,13 +11,13 @@ public boolean isAboveThreshold(SystemCpuRecord record, String metric,
 			String thresholdValue) throws Exception {
 
 		switch (metric) {
-		case "idle":
+		case "%idle":
 			if(record.getIdleCpuUtilPct() == -1d)
 				throw new Exception("Can not compare raw SystemCpuRecord to threshold");
 			else
 				return record.getIdleCpuUtilPct() > Double.parseDouble(thresholdValue);
 
-		case "iowait":
+		case "%iowait":
```

Comment:
This is done to indicate to the person writing the config that the value is
expressed as a percentage.  In the case of iowait it is fairly
straightforward even if we don't explicitly put the "%".  But if you looka
t system memory record processor, the % is important because free could
mean bytes free, KB free, or free as percent of total mem, etc.  I realized
we needed to put % explicitly for things like free memory, and as I was
doing that, I figured we should use it as a convention to make it clear
what type of value that should be given for comparison.  I think it makes
it more user friendly.  That way you know the units of measurement directly
from the parameter name instead of needing to cross check documentation.

On Thu, Jul 23, 2015 at 10:56 AM, Venkata krishnan Sowrirajan <
notifications@github.com> wrote:

> In
> Distiller-server/src/main/java/com/mapr/distiller/server/processors/SystemCpuRecordProcessor.java
> <https://github.com/vsowrirajan/Distiller/pull/6#discussion_r35350983>:
>
> >  			if(record.getIdleCpuUtilPct() == -1d)
> >  				throw new Exception("Can not compare raw SystemCpuRecord to threshold");
> >  			else
> >  				return record.getIdleCpuUtilPct() > Double.parseDouble(thresholdValue);
> >
> > -		case "iowait":
> > +		case "%iowait":
>
> Why it is %iowait? Why do we need to add % to before metric?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/vsowrirajan/Distiller/pull/6/files#r35350983>.
>